src/main/java/com/endava/internship/collections/StudentSet.java

```diff
@@ -1,84 +1,250 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
 
 public class StudentSet implements Set<Student> {
-    @Override
-    public int size() {
-        //TODO
-        return 0;
-    }
+        private Node root;
+        private List<Student> list;
+        private int size = 0;
 
-    @Override
-    public boolean isEmpty() {
-        //TODO
-        return false;
-    }
+        public StudentSet(){
+            list = new LinkedList<>();
+            root = new Node(null); //??
+        }
 
-    @Override
-    public boolean contains(Object o) {
-        //TODO
-        return false;
-    }
+        @Override
+        public int size() {
+            return size;
+        }
 
-    @Override
-    public Iterator<Student> iterator() {
-        //TODO
-        return null;
-    }
+        @Override
+        public boolean isEmpty() {
+            return size == 0;
+        }
 
-    @Override
-    public Object[] toArray() {
-        //TODO
-        return new Object[0];
-    }
+        @Override
+        public boolean contains(Object o) {
+            if (!(o instanceof Student)){
+                return false;
+            }
 
-    @Override
-    public <T> T[] toArray(T[] ts) {
-        //TODO
-        return null;
-    }
+            Node eNode = new Node((Student) o);
+            return search(root, eNode) != null;
+        }
 
-    @Override
-    public boolean add(Student student) {
-        //TODO
-        return false;
-    }
+        private Node search(Node node, Node eNode){
```

Comment:
rename variables to something meaningful please
like
```suggestion
        private Node search(Node sourceNode, Node nodeToSearch){
```